examples/pulse-mpi.py

```diff
@@ -182,27 +190,34 @@ def main(actx_class, ctx_factory=cl.create_some_context, use_logmgr=True,
             ("t_log.max", "log walltime: {value:6g} s")
         ])
 
-    eos = IdealSingleGas()
+    eos = IdealSingleGas(gamma=1.4, gas_const=1.0)
     gas_model = GasModel(eos=eos)
-    vel = np.zeros(shape=(dim,))
+    velocity = np.zeros(shape=(dim,))
+    velocity[0] = 0.1
     orig = np.zeros(shape=(dim,))
-    initializer = Lump(dim=dim, center=orig, velocity=vel, rhoamp=0.0)
-    wall = SymmetryBoundary()
-    boundaries = {BTAG_ALL: wall}
-    uniform_state = initializer(nodes)
-    acoustic_pulse = AcousticPulse(dim=dim, amplitude=1.0, width=.1,
-                                   center=orig)
-    if rst_filename:
-        current_t = restart_data["t"]
-        current_step = restart_data["step"]
-        current_cv = restart_data["cv"]
-        if logmgr:
-            from mirgecom.logging_quantities import logmgr_set_time
-            logmgr_set_time(logmgr, current_step, current_t)
-    else:
-        # Set the current state from time 0
-        current_cv = acoustic_pulse(x_vec=nodes, cv=uniform_state, eos=eos)
+    initializer = Uniform(dim=dim, velocity=velocity)
+    uniform_state = initializer(nodes, eos=eos)
+
+    from mirgecom.initializers import initialize_flow_solution
+    free_stream_cv = initialize_flow_solution(
+        actx, dcoll, gas_model, btag=DTAG_BOUNDARY("inlet"),
+        pressure=1.0, temperature=1.0, velocity=velocity)
+
+    def _inflow_bnd_state_func(dcoll, btag, gas_model, state_minus, **kwargs):
+        return make_fluid_state(cv=free_stream_cv, gas_model=gas_model)
```

Comment:
I think here you want to not `make_fluid_state` repeatedly, so something like....
```suggestion
    inflow_freestream_state = make_fluid_state(cv=free_stream_cv, gas_model=gas_model)
 
    def _inflow_bnd_state_func(dcoll, btag, gas_model, state_minus, **kwargs):
        return inflow_freestream_state
```